postgresql/initdb.d/01_main.sql

```diff
@@ -53,6 +62,8 @@ CREATE TABLE files (
 
     header               TEXT,
     encryption_method    TEXT,
+    key_hash             TEXT REFERENCES encryption_keys(key_hash) ON DELETE SET NULL,
```

Comment:
I'm not sure the "ON DELETE SET NULL" should be there. If we delete a key in the encryption_keys table there should not be any key_hashes in the files table with that key. So if there are, an error should occur.

Reply:
Thanks for the comment @viklund. Do you want to ensure that a key from the `encryption_keys` table cannot be deleted if there are still files in the `files` table that reference it? If so, we should use 

```
key_hash             TEXT REFERENCES encryption_keys(key_hash) ON DELETE RESTRICT,
```
In this case, psql will throw an error if we try to delete a key from the `encryption_keys` table that is still referenced in the `files` table.

Reply:
There shouldn't be any need to delete anything from the `encryption_keys` table, although a nice sanity feature would be to have the restriction in place.

Reply:
Yes, explicitly specifying RESTRICT might not be a bad idea. Not sure what difference it makes compared to not having a "ON DELETE" option.

Reply:
I removed the "ON DELETE" option completely and the behavior of the database operation meets all requirement. 
1. Can insert a row to the `files` table without `key_hash`, i.e. set `key_hash` to NULL
2. Insert a row in the `files` table with the key_hash not matching any record in the `encryption_keys` table will raise an error
3. Delete a key from the `encryption_keys` table that is still referenced in the `files` table will raise an error.

The changes have been implemented in the commit d7565f2a